### ISSUE TITLE
Switch over to fanotify

### ISSUE THREAD

**Issue by yann-morin-1998** [axkibe/lsyncd#284]
Currently, lsync uses inotify to get FS events.

Although inotify is a huge improvement over dnotify, it still suffers from shortcomings. Especially, when there are a lot of events, some may get missed. Also, watches are not recursive, so sub-dirs need to be watched when created. When a sub-dir is created, there is a race window during which new files may get created therein, before lsyncd has a chance to set a watch on that new sub-dir.

In my use-case to use lsyncd, I often have situations where the numbers of watches is reached (because of watching many sub-dirs), and where the event queue overflows (becasue I e.g. delete/create thousands of files in a rapid-fire).

fanotify is a new notification system in Linux, which fixes all these issues (and adds other improvements!).

It sould be nice if lsyncd would add support for fanotify (either as a replacement or as an alternative to inotify.)

Thanks for reading! :-)
Yann E. MORIN.

**Comment by matt-catalyst** [comment]
fanotify has supported move, create, delete, attrib since the 5.1 kernel -- see https://patchwork.kernel.org/patch/10684853/

Can this be reopened?

**Comment by axkibe** [comment]
Fair enough, this is interesting it finally got in. However, I've to say I'm no longer developing on Lsyncd. So unless someone else does it, it's not going to happen. (I'll have in my TODO list to write a formal, nice "goodbye" message for a while now).
